### PR TITLE
Add rand function

### PR DESCRIPTION
**Pull request by carbotaniuman**
This adds in the posix `rand`, `srand`, and `rand_r` functions. We don't get a lot of state to play with here, just 32 bits which really is not enough for anything too serious. I've chosen Mulberry32 here, which seems good enough at statistical tests, is probably going to be fast enough, and most important uses just 32 bits of state.

### CODE REVIEW

**Comment by carbotaniuman** [comment]
So I've realized that just `rand_r` is broken, and `rand` can use more state, so I've opted from a PCG from the `rand` crate. It seems like there's some sort of build error though? I don't see what's wrong, it works locally for me.

File: c-scape/src/rand/mod.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    libc!(libc::rand());
+
+    // POSIX requires that if the user hasn't initialized the RNG, it behaves
+    // as-if srand(1) was called.
+    if *STATE.get() == None {
+        srand(1);
+    }
+
+    // Sample using a uniform distribution
+    (*STATE.get())
```
Comment (bjorn3) [inline c-scape/src/rand/mod.rs:28]: `rand` is supposed to be MT-Safe. Accessing `STATE` from multiple threads at the same time without synchronization is UB.

Comment (sunfishcode) [reply c-scape/src/rand/mod.rs:28]: I think being safe to call in the presence of threads means that two threads can call `rand` at the same time without UB. I think it makes sense to add the lock here.

**Comment by sunfishcode** [comment]
Looks good!

**Pull request merged**